alf/trainers/evaluator.py

```diff
@@ -38,29 +40,31 @@
     defaults=[None] * 4)
 
 
-class AsyncEvaluator(object):
+class Evaluator(object):
     """Evaluate algorithm performance asynchronously in a different process.
 
     Args:
+        config: the training config
         conf_file: path of the config file
-        root_dir: directory for saving summary and checkpoints
-        num_evns: the number of environments for evaluating
-        seed: random seed
     """
 
-    def __init__(self,
-                 conf_file: str,
-                 root_dir: str,
-                 num_envs: int,
-                 seed: Optional[int] = None):
-        ctx = mp.get_context('spawn')
-        self._job_queue = ctx.Queue()
-        self._done_queue = ctx.Queue()
-        self._worker = ctx.Process(
-            target=_worker,
-            args=(self._job_queue, self._done_queue, conf_file, num_envs,
-                  root_dir, seed))
-        self._worker.start()
+    def __init__(self, config: TrainerConfig, conf_file: str):
+        self._async = config.async_eval
+        num_envs = config.num_eval_environments
+        seed = config.random_seed
+        if self._async:
+            ctx = mp.get_context('spawn')
+            self._job_queue = ctx.Queue()
+            self._done_queue = ctx.Queue()
+            self._worker = ctx.Process(
+                target=_worker,
+                args=(self._job_queue, self._done_queue, conf_file, num_envs,
```

Comment:
Can we directly pass `config` to `_worker` instead of letting it parse `conf_file`?

Reply:
config needs to be parsed again in the spawned subprocess anyway.